build/templates/tox-system_tests.ini.mako

```diff
@@ -82,8 +82,8 @@ deps =
     ${module_name}-system_tests: fasteners
     ${module_name}-system_tests: pytest-json
 % if grpc_supported:
-    ${module_name}-system_tests: grpcio
-    ${module_name}-system_tests: protobuf
+    ${module_name}-system_tests: grpcio >= 1.49.1, < 2.0
+    ${module_name}-system_tests: protobuf >= 4.21, < 5.0
 % endif
```

Comment:
Now that the package has extras, shouldn't you use `extras = grpc` (inside an `%if grpc_supported` block) to enable them?

Reply:
Yeah, that's probably a better way to it and it would also test our optional dependencies.

Reply:
Likewise, if the main package depends on `hightime`, do the tests need to explicitly install it?

Reply:
>Likewise, if the main package depends on hightime, do the tests need to explicitly install it?

My philosophy on declaring dependencies is as follows: If you explicitly import something, you should declare an explicit dependency. We explicitly import hightime in our system tests.

Reply:
> > Likewise, if the main package depends on hightime, do the tests need to explicitly install it?
> 
> My philosophy on declaring dependencies is as follows: If you explicitly import something, you should declare an explicit dependency. We explicitly import hightime in our system tests.

Ok, that makes sense.

Reply:
We now depend on .[grpc], instead, so I will mark this resolved.